spec/support/test_contamination.rb

```diff
@@ -0,0 +1,18 @@
+module Spec
+  module Support
+    module TestContamination
+      def self.setup(main)
+        main.method(:include).owner.prepend(patched_include)
+      end
+
+      def self.patched_include
+        Module.new do
+          def include(included)
+            return super if included == UiConstants # TODO: Why is UiConstants included in `main`??? :(
+            raise RuntimeError, "#{included} included in #{self}", caller
```

Comment:
Maybe offer a suggestion:  modules should be included into classes, not into the toplevel main/Object?